robot_skills/src/robot_skills/hero.py

```diff
@@ -109,3 +106,40 @@ def move_to_hmi_pose(self):
         arm.send_joint_goal('arm_out_of_way', 0.0)
         self.base.force_drive(0, 0, rotation_speed, rotation_duration)
         arm.wait_for_motion_done()
+
+    def move_to_pregrasp_pose(self, arm, grasp_target):
+        """
+        This poses the robot for an inspect.
+
+        :param grasp_target: kdl.Frame with the pose of the entity to be grasp.
+        :return: boolean, false if something went wrong.
+        """
+        # calculate the arm_lift_link which must be sent
+        z_over = 0.4  # height the robot should look over the surface
+        z_hh = 0.9  # height of the robots head at z_arm=0
+        torso_to_arm_ratio = 2.0  # motion of arm/motion of torso
+        z_head = grasp_target.z() + z_over
+
+        if z_head < 1.3:
+            # we dont need to do stupid stuff
+            arm.send_joint_trajectory('prepare_grasp')
+            return True
+
+        # saturate the arm lift goal
+        z_arm = (z_head - z_hh) * torso_to_arm_ratio
+        z_arm = min(0.69, max(z_arm, 0.0))
+
+        pose = arm._arm.default_trajectories['prepare_grasp']
+        pose[1][0] = z_arm
+        arm._arm._send_joint_trajectory(pose)
+
+        angle_offset = -math.atan2(arm.base_offset.y(), arm.base_offset.x())
+        self.base.turn_towards(grasp_target.x(), grasp_target.y(), "/map", angle_offset)
+        arm.wait_for_motion_done()
+        while self.base.local_planner.getStatus() != "arrived":
+            rospy.sleep(0.1)
+        return True
+
+    def go_to_driving_pose(self):
+        arm = self.parts['leftArm']
```

Comment:
This is a firm hardcode, you code use the get arm with the carrying_pose as a prerequisite